engine/src/main/java/io/camunda/zeebe/engine/state/migration/MultiInstanceChildActivatedCountMigration.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.migration;
+
+import io.camunda.zeebe.engine.state.ZbColumnFamilies;
+import io.camunda.zeebe.engine.state.immutable.ZeebeState;
+import io.camunda.zeebe.engine.state.mutable.MutableZeebeState;
+
+
+public class MultiInstanceChildActivatedCountMigration implements MigrationTask {
+
+  @Override
+  public String getIdentifier() {
+    return MultiInstanceChildActivatedCountMigration.class.getSimpleName();
+  }
+
+  @Override
+  public boolean needsToRun(final ZeebeState zeebeState) {
+    return !zeebeState.isEmpty(ZbColumnFamilies.ELEMENT_INSTANCE_PARENT_CHILD);
```

Comment:
Either the migration must be idempotent, or we should only run it exactly once. Even if it is idempotent, we should probably only run it if needed to save start-up time for existing clusters with large states.

❌ We should only run this migration once. As soon as it has run for all existing parents in the parentChildColumnFamily of type multi-instance it shouldn't be run again. So how do we check this?
I think the check should be something similar to the verification that's in the migration itself right now:

```java
parentChildColumnFamily.forEach(
        (key, value) -> {
          final long parentElementInstanceKey = key.first().getValue();
          if (parentElementInstanceKey != -1) {
            final var elementInstance = elementInstanceState.getInstance(parentElementInstanceKey);
            if (elementInstance != null
                && elementInstance
                    .getValue()
                    .getBpmnElementType()
                    .equals(BpmnElementType.MULTI_INSTANCE_BODY)) {
// now we know there exists an element instance of type multi-instance that has a child.
// we should also verify that a child has actually activated
// we should also verify that its `childActivatedCount` is still set to 0, which would only be the case for instances in the state that existed before these changes (0 is the default). 
```